src/quoter.ts

```diff
@@ -51,18 +68,19 @@ export abstract class SwapQuoter {
     const swapInterface = options.useQuoterV2 ? new Interface(IQuoterV2.abi) : this.INTERFACE
 
     if (singleHop) {
-      const quoteV2Params = {
+      const quoteV2Params: QuoteParams = {
+        amount: quoteAmount,
+        fee: route.pools[0].fee,
+        sqrtPriceLimitX96: toHex(options?.sqrtPriceLimitX96 ?? 0),
```

Comment:
The order of these parameters matter when making the function call. 

For QuoterV1, the function argument is an array of values